### PR TITLE
fix: only parse arguments when a tool is called

### PR DESCRIPTION
**Pull request by xavidop**
**This pull request is related to:**

- [x] A bug
- [ ] A new feature
- [ ] Documentation
- [ ] Other (please specify)

**I have checked the following:**

- [x] I have read and understood the [contribution guidelines](https://github.com/TheFireCo/genkit-plugins/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/TheFireCo/genkit-plugins/blob/main/CODE_OF_CONDUCT.md);
- [x] I have added new tests (for bug fixes/features);
- [x] I have added/updated the documentation (for bug fixes / features).

**Description:**
This issue fixes a bug when you call a tool in streaming mode. You have to parse only when the tool is being called.

### CODE REVIEW

**Comment by davidoort** [comment]
@all-contributors add @xavidop as contributor for code, tests

File: .all-contributorsrc [hunk 1]
```diff
@@ -43,6 +43,17 @@
...
+      "login": "xavidop",
+      "name": "Xavier Portilla Edo",
+      "avatar_url": "https://avatars.githubusercontent.com/u/4416096?v=4",
+      "profile": "https://xavidop.me",
+      "contributions": [
+        "code",
+        "doc",
+        "test"
+      ]
+    },
```
Comment (davidoort) [inline .all-contributorsrc:56]: this will be added in a separate PR https://github.com/TheFireCo/genkit-plugins/pull/172
```suggestion
```

**Review by davidoort — CHANGES_REQUESTED** [review]

File: .all-contributorsrc (hunk 1 above)
Comment (xavidop) [reply .all-contributorsrc]: @davidoort this is fixed!

**Review by davidoort — APPROVED** [review]
